docs/usage/security/abstract-authentication-middleware.rst

```diff
@@ -1,83 +1,91 @@
+           ) -> AuthenticationResult:
+               # do something here.
+               ...
 
 As you can see, ``authenticate_request`` is an async function that receives a connection instance and is supposed to return
-an :class:`AuthenticationResult <.middleware.authentication.AuthenticationResult>` instance, which is a pydantic model
-that has two attributes:
+an :class:`~.middleware.authentication.AuthenticationResult` instance, which is a
+:doc:`dataclass <python:library/dataclasses>` that has two attributes:
 
-1. ``user``: a non-optional value representing a user. It's typed as ``Any`` so it receives any value, including ``None``.
+1. ``user``: a non-optional value representing a user. It is typed as ``Any`` so it receives any value,
+   including ``None``.
 2. ``auth``: an optional value representing the authentication scheme. Defaults to ``None``.
 
-These values are then set as part of the "scope" dictionary, and they are made available as
+These values are then set as part of the ``scope`` dictionary, and they are made available as
 :attr:`Request.user <.connection.ASGIConnection.user>`
 and :attr:`Request.auth <.connection.ASGIConnection.auth>` respectively, for HTTP route handlers, and
 :attr:`WebSocket.user <.connection.ASGIConnection.user>` and
 :attr:`WebSocket.auth <.connection.ASGIConnection.auth>` for websocket route handlers.
 
-Example: Implementing a JWTAuthenticationMiddleware
----------------------------------------------------
-
-Since the above is quite hard to grasp in the abstract, lets see an example.
+Creating a Custom JWT Authentication Middleware
+-----------------------------------------------
 
-We start off by creating a user model. It can be implemented using pydantic, and ODM, ORM, etc. For the sake of the
-example here lets say it's a SQLAlchemy model:
+Since the above is quite hard to grasp in the abstract, let us see an example.
 
-.. code-block:: python
-    :caption: my_app/db/models.py
+We start off by creating a user model. It can be implemented using Pydantic, an ODM, ORM, etc. For the sake of the
+example here let us say it is a `SQLAlchemy <https://docs.sqlalchemy.org/>`_ model:
 
-    import uuid
+.. dropdown:: Click to see the User model
 
-    from sqlalchemy import Column
-    from sqlalchemy.dialects.postgresql import UUID
-    from sqlalchemy.orm import declarative_base
+    .. code-block:: python
+        :caption: my_app/db/models.py
 
-    Base = declarative_base()
+        import uuid
 
+        from sqlalchemy import Column
+        from sqlalchemy.dialects.postgresql import UUID
+        from sqlalchemy.orm import declarative_base
 
-    class User(Base):
-        id: uuid.UUID | None = Column(
-            UUID(as_uuid=True), default=uuid.uuid4, primary_key=True
-        )
-        # ... other fields follow, but we only require id for this example
+        Base = declarative_base()
 
 
+        class User(Base):
+            id: uuid.UUID | None = Column(
```

Comment:
should we use SQLA 2.x ``mapped_column`` so that we are consistent with the other examples where we use SQLA models?

Reply:
yes